doc/guides/debugging/index.rst

```diff
@@ -8,3 +8,4 @@ Debugging
 
    host-tools.rst
    probes.rst
+   stlink-v21.rst
```

Comment:
Is this separate page really still necessary?

Reply:
Information provided in this page is not available anywhere else so it's worth keeping, IMHO. This being said I'm open to alternative locations.

Reply:
I think it can be worked into the debug probe page

Reply:
Ok, done!